state.c

```diff
@@ -411,20 +414,66 @@ state_spec_t pv_state_spec(struct pv_state *s)
 	return s->spec;
 }
 
+static char* _pv_state_get_novalidate_list(char *rev)
+{
+#define _pv_bufsize_1 (1288 * 1024)
+#define _cmd_fmt "/lib/pv/volmount/dm noverifylist %s"
+	char *path = pv_storage_get_rev_path(rev);
+	char sout[_pv_bufsize_1] = {0}, serr[_pv_bufsize_1] = {0};
+	char *cmd = malloc(sizeof(char) * (strlen(_cmd_fmt) + strlen(path) + 1));
+	int res;
+
+	sprintf(cmd, _cmd_fmt, path);
+	pv_log(DEBUG, "running command: %s", cmd);
+
+	res = tsh_run_output(cmd, 20, sout, _pv_bufsize_1 - 1, serr, _pv_bufsize_1 - 1);
+
+	if (res < 0) {
+		pv_log(WARN, "error running validate_list command: %s", strerror(errno));
+		return NULL;
+	}
+	if (res > 0) {
+		pv_log(WARN, "command exited with error code: %d", res);
+		pv_log(DEBUG, "stdout: %s", sout);
+		pv_log(DEBUG, "stderr: %s", serr);
+		return NULL;
+	}
+
+	if (strlen(serr) > 0)
+		pv_log(WARN, "get_novalidate_list stderr output: %s", stderr);
+
+	return strndup(sout, _pv_bufsize_1 - 1);;
+}
+
+#ifndef ARRAY_SIZE
+#define ARRAY_SIZE(a) (sizeof(a)/sizeof((a)[0]))
+#endif
+
 bool pv_state_validate_checksum(struct pv_state *s)
 {
 	struct pv_object *o;
 	struct pv_json *j;
 
+	char *validate_list = _pv_state_get_novalidate_list(s->rev);
+	if (validate_list)
+		pv_log(DEBUG, "no validation list is: %s", validate_list);
+
 	pv_objects_iter_begin(s, o) {
 		/* validate instance in $rev/trails/$name to match */
-		if (!pv_storage_validate_trails_object_checksum(s->rev, o->name, o->id)) {
-			pv_log(ERROR, "trails object %s with checksum %s failed", o->name, o->id);
+		char needle[PATH_MAX + 67];
```

Comment:
why PATH_MAX + 67?

Reply:
its a path + an ID which i guess could be smaller but left it at this for now...